File: java/vetores/MediaPares.java

import entities.ExercicioMediaPares;

import java.util.Scanner;

public class MediaPares {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        
        int i, N, quant, valor;

        quant = 0;

        System.out.print("Quantos numeros voce vai digitar? ");
        N = sc.nextInt();

        ExercicioMediaPares[] num = new ExercicioMediaPares[N];

        for (i = 0; i < N; i++) {
            System.out.print("Digite um numero: ");
            valor = sc.nextInt();
            num[i] = new ExercicioMediaPares(valor);
        }

        for (i = 0; i < N; i++) {
            if (num[i].getNum() % 2 == 0) {
                quant++;
            }
        }

        System.out.println("NUMEROS PARES: ");
        for (i = 0; i < N; i++) {
            if (num[i].getNum() % 2 == 0) {
                System.out.print(num[i].getNum() + " ");
            }
        }
        System.out.println();
        System.out.printf("QUANTIDADE DE PARES = %d", quant);
    }
}

File: js/vetores/mediaPares.js
const readline = require('readline-sync');

const X = Number(readline.question("Quantos elementos vai ter o vetor? "));
const numeros = [];

for (let i = 0; i < X; i++) {
    const num = Number(readline.question("Digite um numero: "));
    numeros.push(num);
}

const somaPares = numeros.filter(num => num % 2 === 0).reduce((acc, val) => acc + val, 1);
const pares = numeros.filter(num => num % 2 === 0);


if (pares.length > 0) {
    const media = somaPares / pares.length;
    console.log(`MEDIA DOS PARES = ${media.toFixed(1)}`);
} else {
    console.log("NENHUM NUMERO PAR");
}